gobblin-cluster/src/main/java/org/apache/gobblin/cluster/SingleHelixTask.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.gobblin.runtime.AbstractJobLauncher;
+import org.apache.gobblin.runtime.GobblinMultiTaskAttempt;
+import org.apache.gobblin.runtime.JobState;
+import org.apache.gobblin.runtime.util.StateStores;
+import org.apache.gobblin.source.workunit.MultiWorkUnit;
+import org.apache.gobblin.source.workunit.WorkUnit;
+import org.apache.gobblin.util.JobLauncherUtils;
+import org.apache.gobblin.util.SerializationUtils;
+
+
+public class SingleHelixTask {
+
+  private static final Logger _logger = LoggerFactory.getLogger(SingleHelixTask.class);
+
+  private GobblinMultiTaskAttempt taskAttempt;
+  private String jobId;
+  private Path workUnitFilePath;
+  private Path jobStateFilePath;
+  private FileSystem fs;
+  private TaskAttemptBuilder taskAttemptBuilder;
+  private StateStores stateStores;
+
+  public SingleHelixTask(String jobId, Path workUnitFilePath, Path jobStateFilePath, FileSystem fs,
+      TaskAttemptBuilder taskAttemptBuilder, StateStores stateStores) {
+    this.jobId = jobId;
+    this.workUnitFilePath = workUnitFilePath;
+    this.jobStateFilePath = jobStateFilePath;
+    this.fs = fs;
+    this.taskAttemptBuilder = taskAttemptBuilder;
+    this.stateStores = stateStores;
+  }
+
+  public int run()
+      throws IOException, InterruptedException {
+    SharedResourcesBroker<GobblinScopeTypes> globalBroker = null;
+    try {
+      return runInternal(globalBroker);
+    }finally {
+      closeGlobalBroker(globalBroker);
+    }
+  }
+
+  private int runInternal(SharedResourcesBroker<GobblinScopeTypes> globalBroker)
+      throws IOException, InterruptedException {
+    List<WorkUnit> workUnits = getWorkUnits();
+    int workUnitSize = workUnits.size();
+
+    JobState jobState = getJobState();
+
+    globalBroker = SharedResourcesBrokerFactory
+        .createDefaultTopLevelBroker(ConfigFactory.parseProperties(jobState.getProperties()), GobblinScopeTypes.GLOBAL.defaultScopeInstance());
+    SharedResourcesBroker<GobblinScopeTypes> jobBroker =
+        globalBroker.newSubscopedBuilder(new JobScopeInstance(jobState.getJobName(), jobState.getJobId())).build();
+
+    this.taskAttempt = this.taskAttemptBuilder.build(workUnits.iterator(), this.jobId, jobState, jobBroker);
+    this.taskAttempt.runAndOptionallyCommitTaskAttempt(GobblinMultiTaskAttempt.CommitPolicy.IMMEDIATE);
+    return workUnitSize;
+
+  }
+  private void closeGlobalBroker(SharedResourcesBroker<GobblinScopeTypes> globalBroker) {
+    if (globalBroker != null) {
+      try {
+        globalBroker.close();
+      } catch (IOException ioe) {
+        _logger.error("Could not close shared resources broker.", ioe);
+      }
+    }
+  }
+
+  private JobState getJobState()
+      throws java.io.IOException {
+    JobState jobState = new JobState();
+    SerializationUtils.deserializeState(this.fs, jobStateFilePath, jobState);
+    return jobState;
+  }
+
+  private List<WorkUnit> getWorkUnits()
+      throws IOException {
```

Comment:
Is this a coding style that exceptions will be a new line?

Reply:
IDE auto formatted it based on the style config Gobblin provides.